src/hooks/useCountries.js

```diff
@@ -0,0 +1,34 @@
+import { useEffect, useState } from 'react';
+import { fetchCountries } from '../services/countries';
+
+export function useCountries() {
+  const [countries, setCountries] = useState([]);
+  const [continent, setContinent] = useState('all');
+  const [error, setError] = useState('');
+  const [query, setQuery] = useState('');
+  const [loading, setLoading] = useState(true);
+
+  useEffect(() => {
+    async function fetchData() {
+      try {
+        const data = await fetchCountries();
+        setCountries(data);
+        setTimeout(() => setLoading(false), 2000);
+      } catch (e) {
+        setError(e.message.toUpperCase());
+      }
+    }
+    fetchData();
+  }, []);
+
+  const filterCountries = () => {
+    const filteredCountries = countries.filter(
+      (country) =>
+        (continent === 'all' ? true : country.continent === continent) &&
+        country.name.toLowerCase().includes(query)
+    );
+    return filteredCountries;
+  };
```

Comment:
In general I'd recommend avoiding the use of ternary operators when they are nested inside other expressions like this. What you did works, but it is a little difficult to read and took me a moment to piece together what was happening. In this case you can replace the ternary with the [logical OR](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Logical_OR) ( || ) to clean it up a bit and improve readability. 

```js
const filteredCountries = countries.filter(country => {
  return (continent === 'all' || country.continent === continent) &&
    country.name.includes(query);
});